### PR TITLE
ja: : cr2 Day 1 translations

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all Japanese content is reviewed.

#1463 #652

### CODE REVIEW

**Comment by kantasv** [comment]
Hi @henrif75 , thanks for raising the MR. I'm taking a look and will get back to you once I finish review :)

File: po/ja.po [hunk 1]
```diff
@@ -3475,339 +3632,369 @@ msgid ""
...
 
 #: src/tuples-and-arrays/tuples-and-arrays.md
 msgid "`()`, `('x',)`, `('x', 1.2)`, ..."
-msgstr ""
+msgstr "`()`、`('x',)`、`('x', 1.2)`、..."
 
 #: src/tuples-and-arrays/tuples-and-arrays.md
 msgid "Array assignment and access:"
-msgstr ""
+msgstr "配列の割り当てとアクセス権:"
```
Comment (kantasv) [inline po/ja.po:3989]:
```suggestion
msgstr "配列の代入とアクセス:"
```

File: po/ja.po [hunk 2]
```diff
@@ -3475,339 +3632,369 @@ msgid ""
...
 
 #: src/tuples-and-arrays/tuples-and-arrays.md
 msgid "Tuple assignment and access:"
-msgstr ""
+msgstr "タプルの割り当てとアクセス権:"
```
Comment (kantasv) [inline po/ja.po:3993]:
```suggestion
msgstr "タプルの代入とアクセス:"
```

File: po/ja.po [hunk 3]
```diff
@@ -3817,17 +4004,24 @@ msgid ""
...
 "avoided using unsafe Rust."
 msgstr ""
+"境界外の配列要素にアクセスしてみてください。配列アクセスは実行時にチェックさ"
+"れます。Rust では通常、これらのチェックを最適化により除去できます（安全でな"
+"い Rust を使用することで回避できます）。"
 
 #: src/tuples-and-arrays/tuples-and-arrays.md
 msgid "We can use literals to assign values to arrays."
-msgstr ""
+msgstr "リテラルを使用して配列に値を割り当てることができます。"
```
Comment (kantasv) [inline po/ja.po:4024]:
```suggestion
msgstr "リテラルを使用して配列に値を代入することができます。"
```

File: po/ja.po [hunk 4]
```diff
@@ -3869,118 +4071,141 @@ msgid ""
...
 
 #: src/tuples-and-arrays/match.md
 msgid ""
 "The `_` pattern is a wildcard pattern which matches any value. The "
 "expressions _must_ be irrefutable, meaning that it covers every possibility, "
 "so `_` is often used as the final catch-all case."
 msgstr ""
+"`_` パターンは、任意の値に一致するワイルドカード パターンです。式は不可反駁で"
+"なければなりません。つまり、あらゆる可能性をカバーする必要があるため、最後の"
+"キャッチオール ケースとして `_` がよく使用されます。"
```
Comment (kantasv) [inline po/ja.po:4186]:
```suggestion
"`_` パターンは、任意の値に一致するワイルドカード パターンです。式は論駁不可能で"
"なければなりません。つまり、あらゆる可能性をカバーする必要があるため、最後の"
"キャッチオール ケースとして `_` がよく使用されます。"
```

irrefutable -> 論駁不可能, based on [existing Rustbook JA translation](https://doc.rust-jp.rs/book-ja/ch18-02-refutability.html)

**Review by kantasv — COMMENTED** [review]
Hi @henrif75 , I've skimmed thru the diff and added suggestion by Github feature. You can incorperate changes from Github GUI although you might run dprint again to pass CI. 

I have spotted clear translation suggestions. After my review, other folks might add in-depth reviews but I try to spot clear ones to offload other reviewers.

**Comment by henrif75** [comment]
> Hi @henrif75 , I've skimmed thru the diff and added suggestion by Github feature. You can incorperate changes from Github GUI although you might run dprint again to pass CI.
> 
> I have spotted clear translation suggestions. After my review, other folks might add in-depth reviews but I try to spot clear ones to offload other reviewers.

Thank you @kantasv !

File: po/ja.po [hunk 5]
```diff
@@ -2799,194 +2877,215 @@ msgid ""
...
 "Rust provides type safety via static typing. Variable bindings are made with "
 "`let`:"
 msgstr ""
+"Rust は静的型付けによって型安全性を提供します。変数のバインディングは `let` "
+"を使用して行います。"
 
 #: src/types-and-values/variables.md src/control-flow-basics/loops.md
 #: src/control-flow-basics/break-continue.md
 #: src/control-flow-basics/blocks-and-scopes.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:2924]: remove fuzzy

File: po/ja.po [hunk 6]
```diff
@@ -3253,15 +3391,19 @@ msgid ""
...
+"必要があります。`println!` の前にある `;` を削除すると、コンパイラ エラーが表"
+"示されます。"
 
 #: src/control-flow-basics/loops.md
 msgid "There are three looping keywords in Rust: `while`, `loop`, and `for`:"
 msgstr ""
+"Rust には、`while`、`loop`、`for` の 3 つのループ キーワードがあります。"
 
 #: src/control-flow-basics/loops.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:3404]: Remove fuzzy

File: po/ja.po [hunk 7]
```diff
@@ -3253,15 +3391,19 @@ msgid ""
...
 msgid "`while`"
-msgstr "`while` ループ"
+msgstr "while-let"
 
 #: src/control-flow-basics/loops.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:3409]: Remove fuzzy

File: po/ja.po [hunk 8]
```diff
@@ -2603,7 +2658,7 @@ msgstr ""
 #: src/hello-world/hello-world.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:2659]: Remove fuzzy

File: po/ja.po [hunk 9]
```diff
@@ -1751,7 +1760,7 @@ msgstr ""
 #: src/running-the-course/course-structure.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:1761]: Remove fuzzy

File: po/ja.po [hunk 10]
```diff
@@ -725,24 +716,24 @@ msgstr "テストモジュール"
...
-msgstr ""
+msgstr "生成された C++"
 
 #: src/SUMMARY.md
 msgid "C++ Bridge"
-msgstr ""
+msgstr "C++ ブリッジ"
 
 #: src/SUMMARY.md src/android/interoperability/cpp/shared-types.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:730]: Remove fuzzy

File: po/ja.po [hunk 11]
```diff
@@ -725,24 +716,24 @@ msgstr "テストモジュール"
 #: src/SUMMARY.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:717]: Remove fuzzy

File: po/ja.po [hunk 12]
```diff
@@ -643,7 +634,7 @@ msgstr "Unsafeなトレイトの実装"
...
```
Comment (henrif75) [inline po/ja.po:635]: Remove fuzzy

File: po/ja.po [hunk 13]
```diff
@@ -451,12 +445,12 @@ msgstr "借用"
 #: src/SUMMARY.md src/borrowing/interior-mutability.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:446]: Remove

File: po/ja.po [hunk 14]
```diff
@@ -356,20 +351,20 @@ msgstr "テスト"
...
-msgstr ""
+msgstr "`Default`、構造体更新構文"
 
 #: src/SUMMARY.md src/std-traits/closures.md
 msgid "Closures"
-msgstr ""
+msgstr "クロージャ"
 
 #: src/SUMMARY.md src/std-traits/exercise.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:365]: Remove

File: po/ja.po [hunk 15]
```diff
@@ -301,35 +298,33 @@ msgid "Standard Library"
...
-msgstr ""
+msgstr "`Vec`"
 
 #: src/SUMMARY.md src/std-types/hashmap.md src/bare-metal/no_std.md
 msgid "`HashMap`"
-msgstr ""
+msgstr "`HashMap`"
 
 #: src/SUMMARY.md src/std-types/exercise.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:325]: Remove

File: po/ja.po [hunk 16]
```diff
@@ -260,7 +257,7 @@ msgstr "トレイトオブジェクト"
 #: src/SUMMARY.md src/methods-and-traits/exercise.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:258]: Remove

File: po/ja.po [hunk 17]
```diff
@@ -154,12 +153,12 @@ msgstr "Day 1： PM"
 #: src/SUMMARY.md src/tuples-and-arrays.md
 #: src/tuples-and-arrays/tuples-and-arrays.md
 msgid "Tuples and Arrays"
-msgstr ""
+msgstr "タプルと配列"
 
 #: src/SUMMARY.md src/tuples-and-arrays/iteration.md
 #, fuzzy
```
Comment (henrif75) [inline po/ja.po:159]: Remove

File: po/ja.po [hunk 18]
```diff
@@ -53,52 +53,53 @@ msgid "Day 1: Morning"
 msgstr "Day 1： AM"
 
 #: src/SUMMARY.md
+#, fuzzy
```
Comment (henrif75) [inline po/ja.po:56]: Remove

File: po/ja.po [hunk 19]
```diff
@@ -356,20 +351,20 @@ msgstr "テスト"
 
 #: src/SUMMARY.md src/std-traits/read-and-write.md
 msgid "`Read` and `Write`"
-msgstr ""
+msgstr "`Read` と `Write`"
 
 #: src/SUMMARY.md
 msgid "`Default`, struct update syntax"
-msgstr ""
+msgstr "`Default`、構造体更新構文"
```
[the remark this answers is not in the document]
Comment (kantasv) [reply po/ja.po:354]: ### 
```suggestion
msgstr "`Default`、構造体更新記法"
```

Followed translation from [Rustbook JA](https://doc.rust-jp.rs/book-ja/ch05-01-defining-structs.html#%E6%A7%8B%E9%80%A0%E4%BD%93%E6%9B%B4%E6%96%B0%E8%A8%98%E6%B3%95%E3%81%A7%E4%BB%96%E3%81%AE%E3%82%A4%E3%83%B3%E3%82%B9%E3%82%BF%E3%83%B3%E3%82%B9%E3%81%8B%E3%82%89%E3%82%A4%E3%83%B3%E3%82%B9%E3%82%BF%E3%83%B3%E3%82%B9%E3%82%92%E7%94%9F%E6%88%90%E3%81%99%E3%82%8B)

**Review by kantasv — COMMENTED** [review]
Hi @henrif75 , thanks for fixing the po file. I added a suggestion per "struct update syntax".

**Review by kantasv — APPROVED** [review]
Thanks @henrif75 . Overall, LGTM!

**Comment by henrif75** [comment]
> Thanks @henrif75 . Overall, LGTM!

Thank you @kantasv !

**Pull request merged**